controllers/users-controller.js

```diff
@@ -0,0 +1,80 @@
+import { Request, Response, NextFunction } from 'express';
+import bcrypt from 'bcryptjs';
+import createError from 'http-errors';
+import jwt from 'jsonwebtoken';
+import models from '../models';
+import { env } from '../helpers/utils';
+
+const generateToken = async payload => jwt.sign(payload, env('APP_KEY'));
+
+const { User } = models;
+
+/**
+ * Class representing users controller
+ *
+ * @class UsersController
+ */
+class UsersController {
+  /**
+   * Creates a new user resource
+   *
+   * @static
+   * @param {Request} request - Request object
+   * @param {Response} response - Response object
+   * @param {NextFunction} next - Express next function
+   * @returns {void}
+   *
+   * @memberof UsersController
+   */
+  static async index(request, response, next) {
+    const { body } = request;
+
+    try {
+      const user = await User.create(body);
+      const token = await generateToken({ user });
+
+      response
+        .status(201)
+        .json({ token, id: user.id });
+    } catch (error) {
+      next(error);
+    }
+  }
+
+  /**
+   * Authenticate a user
+   *
+   * @static
+   * @param {Request} request - Request object
+   * @param {Response} response - Response object
+   * @param {NextFunction} next - Express next function
+   * @returns {void}
+   *
+   * @memberof UsersController
+   */
+  static async login(request, response, next) {
```

Comment:
#### Nitpick 
- Any reason why you choose to make the functions static?

Reply:
So that method can be accessible using the convention `UsersController.login` without specifically instantiating the controller.